tsconfig.json

```diff
@@ -11,6 +11,7 @@
     "allowSyntheticDefaultImports": true,
     "sourceMap": true,
     "baseUrl": ".",
+    "strictPropertyInitialization": false,
```

Comment:
What effect does this have exactly?

Reply:
Well, this is more like a proposal than a must have change.

here is the definition from the official doc https://www.typescriptlang.org/tsconfig#strictPropertyInitialization
> TypeScript will raise an error when a class property was declared but not set in the constructor.

Why do we need it?
Our current code has TS error like this
```
ERROR in /home/safydy/Documents/flow-builder/src/components/interaction-designer/resource-editors/ResourceEditor.vue(131,29):
131:29 Property 'block' has no initializer and is not definitely assigned in the constructor.
    129 |   })
    130 | export class ResourceEditor extends mixins(FlowUploader, Permissions, Routes, Lang) {
  > 131 |   @Prop({ required: true }) block: IBlock
           |                                            ^
    132 | 
    133 |   @Prop({ required: true }) flow: IFlow
    134 | 
ERROR in /home/safydy/Documents/flow-builder/src/components/interaction-designer/resource-editors/ResourceEditor.vue(133,29):
133:29 Property 'flow' has no initializer and is not definitely assigned in the constructor.
    131 |   @Prop({ required: true }) block: IBlock
    132 | 
  > 133 |   @Prop({ required: true }) flow: IFlow
           |                                            ^
    134 | 
    135 |   @Prop() resource: IResourceDefinition
```

The solution is to provide initial value `@Prop({ required: true, default: {} }) block: IBlock = {} as IBlock`, but it's too repetitive to my POV. 
As I never had an issue without having initialized a prop in the past, I thought we could ignore this kind of error from the tsconfig. What do you think?

Reply:
How about this instead? 

https://github.com/kaorun343/vue-property-decorator/issues/81#issuecomment-365941684

The issue with this approach is that I think it is hiding that resource and label are not required/don't have a default. We should make sure we set these in all cases.

Reply:
BTW. I suggest we don't prioritise this ticket for the next sprint while we still have functional changes to do. We won't be able to merge until flow lifecycle is merged anyway.

Reply:
@jcbashdown , thanks for the suggestion. Let's adopt the `definite assignment assertion` then. I have remove the tsconfig `"strictPropertyInitialization": false`.
About priority, yes we should prioritize flow lifecycle as discussed, but upcoming functionalities (eg: Validation API tickets: vmo-3710, vmo-3704, etc) were branched out of this to avoid headache on conflict resolution. Our plan is to merge this asap once flow lifecycle is merged.